.github/workflows/ci.yml

```diff
@@ -15,27 +15,80 @@ jobs:
   sensuflow:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
-
+    container: ubuntu
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - name: Checkout
       uses: actions/checkout@v2
+    - name: prep apt
+      run: >-
+        apt update && apt install -y
+        apt-transport-https 
+        ca-certificates 
+        curl 
+        gnupg 
+        net-tools
+        lsb-release
+    - name: get docker gpg
+      run: >-
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+    - name: setup docker apt source
+      run: >-
+        echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" 
+        | tee /etc/apt/sources.list.d/docker.list > /dev/null
+    - name: install docker
+      run: >-
+        DEBIAN_FRONTEND=noninteractive apt update && apt install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+    # Run a locally available sensu-backend service from a docker container
+    - name: Run local sensu-backend using docker
+      run: >-
+        docker run 
+        -d
+        --name sensu-backend
+        --hostname sensu-backend 
+        --network ${{ job.container.network }}  
+        -e SENSU_BACKEND_CLUSTER_ADMIN_USERNAME=${{ secrets.SENSU_USER }} 
+        -e SENSU_BACKEND_CLUSTER_ADMIN_PASSWORD=${{ secrets.SENSU_PASSWORD }}
+        sensu/sensu:latest
+        sensu-backend start
+    - name: wait for sensu-backend
+      run: curl --head -X GET --retry 10 --retry-connrefused --retry-delay 2 http://sensu-backend:8080/health
+    - name: docker ps
+      run: docker ps
+    - name: docker logs
+      run: docker logs sensu-backend
+    - name: sensuctl configure
+      run: docker exec sensu-backend sensuctl configure --api-url http://localhost:8080 --username  ${{ secrets.SENSU_USER }} --password  ${{ secrets.SENSU_PASSWORD }} -n
+    - name: grant api key
+      run: echo "::set-output name=SENSU_API_KEY::$(basename $(docker exec sensu-backend sensuctl api-key grant ${{ secrets.SENSU_USER }} | awk -F ':' '{print $2}'))"
+      id: grant-api-key
+    - name: check api key
+      run: echo "The api key is ${{ steps.grant-api-key.outputs.SENSU_API_KEY }}"
```

Comment:
You could skip all of these steps by adding a SENSU_BACKEND_CLUSTER_ADMIN_APIKEY like we do in the workshop now. 

See: https://github.com/sensu/sensu-go-workshop/blob/latest/docker-compose-default.yaml#L26